### PR TITLE
Patches for a single package plus errata details infrastructure

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
This PR implements a context function to retrieve relevant patches covered by a package upgrade, needed by the upgradable packages overview. In addition to that, I also felt free to implement the ground work to get the specific details for a relevant patch, to be used in the detailed view of a relevant patch itself.

I'm really sorry for messing up the reviewability of this a little bit, but I meant no harm :crying_cat_face: 

## How was this tested?

Existing tests passing, unit tests added

### CODE REVIEW

File: lib/trento/infrastructure/software_updates/suma_api.ex [hunk 1]
```diff
@@ -183,19 +294,24 @@ defmodule Trento.Infrastructure.SoftwareUpdates.SumaApi do
...
+            ]
+          }}
+       ) do
+    {:ok, system_id}
   end
 
+  defp extract_system_id({:ok, _}), do: {:error, :system_id_not_found}
+
+  defp extract_system_id({:error, _} = error), do: error
+
```
Comment (janvhs) [inline lib/trento/infrastructure/software_updates/suma_api.ex:315]: It becomes way more visible what is expected. Love it, but it seems like you stripped the logging. Was that a conscious decision?

**Review by janvhs — APPROVED** [review]
Tried following along and LGTM

File: lib/trento/infrastructure/software_updates/suma_api.ex (hunk 1 above)
Comment (dottorblaster) [reply lib/trento/infrastructure/software_updates/suma_api.ex:315]: You're actually right, it was absolutely unconscious, changing back :smile:

File: test/trento/infrastructure/software_updates/suma_test.exs [hunk 2]
```diff
@@ -194,13 +194,161 @@ defmodule Trento.Infrastructure.SoftwareUpdates.SumaTest do
...
+
+      expect(SumaApiMock, :get_errata_details, 1, fn _, _, ^advisory_name, _ ->
+        {:ok, %HTTPoison.Response{status_code: 200, body: Jason.encode!(suma_response_body)}}
+      end)
+
+      assert {:ok, ^errata_details} =
+               Suma.get_errata_details(advisory_name)
+    end
+
+    test "should return a proper error when getting " do
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply test/trento/infrastructure/software_updates/suma_test.exs:339]: D:

**Review by jamie-suse — APPROVED** [review]

**Pull request merged**